### ISSUE TITLE
Appending a value to a not yet initialized metadata lead to a 500 error

### ISSUE THREAD

**Issue by abollini** [DSpace/DSpace#3377]
**Describe the bug**
This error is only reproducible by direct interaction with the REST API not using the current UI.
If you try to append a new value to a metadata that is not yet in use (i.e. append an author when there is not yet any author) the server return a 500 error.

**To Reproduce**
1. Create a new workspace item in a standard collection
2. Try to add an author using this json patch ```[
  {"op":"add","path":"/sections/traditionalpageone/dc.contributor.author/-", "value": {"value": "Test, Author"}}
]```
3. you will see the error. Please note that the previous json patch works properly once that a first author is created for instance with this json patch ```[
  {"op":"add","path":"/sections/traditionalpageone/dc.contributor.author", "value": [{"value": "First, Author"}]}
]```

**Expected behavior**
Not 100% sure but probably we should be allow the client to make such request and initialize the author metadata with the provided value. This seems to be inline with the json patch RFC but it is not so explicit
https://datatracker.ietf.org/doc/html/rfc6902#section-4.1

> Because this operation is designed to add to existing objects and  arrays, its target location will often not exist.  Although the
   pointer's error handling algorithm will thus be invoked, this specification defines the error handling behavior for "add" pointers
   to ignore that error and add the value as specified. 
> However, the object itself or an array containing it does need to exist, and it remains an error for that not to be the case.  ...

If we decide to be friendly with the client I would suggest to process both an append json patch request `/-` than a `/0` request keeping rejecting with a 422 any request that refer to an index that doesn't yet exist and will be not created by the patch itselt (i.e. `/1` or `/2` etc.).

If we decide to reject such request at least we should return a 422 error code instead than an internal error message.

**Comment by tdonohue** [comment]
@abollini : I agree that it makes sense to not throw an error here, and accept an `add` PATCH request to `/-` or `/0` for a field that doesn't yet have values.  That said, this doesn't seem high priority to me (unless it is necessary for an Angular feature or similar).  So, while I've moved it to 7.1, it's flagged as `low priority` for now.  This means, if you wanted to implement as a **volunteer** , you are welcome to do so, but it won't be paid work until higher priority tasks are completed.
